### PR TITLE
Reset portion size before dumping in hopper controller

### PR DESCRIPTION
**Pull request by luke-powers**
Reset to the smallest portion size after dump in order to ensure that
only the smallest amount is dispensed so if a request for a larger
size comes in, the hopper can still be used (simply add more product
until the larger size weight is achieved).

Other minor bug fixes as well.

### CODE REVIEW

File: miso_peripherals/src/miso_peripherals/hopper/ram_driver.py [hunk 1]
```diff
@@ -35,12 +35,13 @@ def __init__(self, device_driver, hopper):
...
     def dump(self):
         """Dump food from the hopper."""
+        self._reset_desired_portion_size()
         self._device_driver.rising_edge(
             self._pin_config.dump,
             PULSE_DURATION,
         )
 
-    def refill(self):
+    def refill(self, weight):
```
Comment (rsinnet) [inline miso_peripherals/src/miso_peripherals/hopper/ram_driver.py:44]: oof

File: miso_peripherals/src/miso_peripherals/hopper/sim_driver.py [hunk 2]
```diff
@@ -106,6 +106,7 @@ def dump(self):
             float: The amount dispensed.
         """
         with self._lock:
+            self._reset_desired_portion_size()
```
Comment (rsinnet) [inline miso_peripherals/src/miso_peripherals/hopper/sim_driver.py:109]: Nice.

File: miso_peripherals/test/test_ram_hopper.py [hunk 3]
```diff
@@ -99,3 +99,10 @@ def test_bin(self):
 
         self.assertEquals(0, bin_.dispense(1.))
         self.assertTrue(bin_.empty)
+
+    def test_reset_after_dump(self):
+        hopper = self._hopper
+        hopper.desired_portion_size = SetPortionSizeRequest.LARGE
+        hopper.dump()
```
Comment (rsinnet) [inline miso_peripherals/test/test_ram_hopper.py:106]: You could verify the setter worked here. Then you could also validate the `actual_portion` is equal to what you set before and after setting it.

**Review by rsinnet — APPROVED** [review]
Awesome work. A couple comments on additional test conditions to consider.

File: miso_peripherals/test/test_ram_hopper.py (hunk 3 above)
Comment (luke-powers) [reply miso_peripherals/test/test_ram_hopper.py:116]: Updated.

File: miso_peripherals/test/test_sim_hopper.py [hunk 4]
```diff
@@ -52,7 +52,7 @@ def setUp(self):
             HopperControllerNode.REFILL_SRV,
             HopperRefill,
         )
-        self._portion = rospy.ServiceProxy(
+        self.desired_portion_size = rospy.ServiceProxy(
```
Comment (rsinnet) [inline miso_peripherals/test/test_sim_hopper.py:55]: Looks like you missed an underscore.

File: miso_peripherals/test/test_ram_hopper.py (hunk 3 above)
Comment (rsinnet) [reply miso_peripherals/test/test_ram_hopper.py:116]: Looks great!

File: miso_peripherals/test/test_sim_hopper.py (hunk 4 above)
Comment (luke-powers) [reply miso_peripherals/test/test_sim_hopper.py:55]: `desired_portion_size` is the property that changes `_desired_portion_size`

Comment (rsinnet) [reply miso_peripherals/test/test_sim_hopper.py:55]: Look this variable holds a `ServiceProxy`, not a portion size.

Comment (luke-powers) [reply miso_peripherals/test/test_sim_hopper.py:55]: Ah I see, this is on the test object, not a Hopper object. I'll change it back.

**Review by pramati — APPROVED** [review]

**Review by zzv2 — APPROVED** [review]

**Comment by luke-powers** [comment]
Fixed up and rebased to develop.

**Pull request merged**